### ISSUE TITLE
Connector snapping effect doesn't happen anymore when it should

### ISSUE THREAD

**Issue by strblr** [xyflow/xyflow#3378]
### Describe the Bug

Since this [release](https://github.com/wbkd/react-flow/releases/tag/11.7.3), the effect of the connector snapping to the node doesn't work properly anymore. I believe it is related to this update:

> Use all valid handles in proximity to determine the closest handle that can be snapped to. Fixes invalid handles connection radius blocking snap to valid handles inside the same radius

### Steps to Reproduce the Bug or Issue

1. Render a big target handle behind a node to make it connectable from all sides
2. Approach a connector line towards a node
3. It snaps to it at the edge, but then if the mouse goes further inside the node, the snapping effect stops

### Expected behavior

Even when hovering over a node and not directly the handle, the handle is still in the connection radius and should trigger a snap effect, fixing `toX` and `toY` in custom connectors.

### Screenshots or Videos

![CPT2309102311-879x528](https://github.com/wbkd/react-flow/assets/30187797/c2ec20f9-a5ad-42ce-9875-32577674c6aa)

As you can see, the snap effect only happens when we're really at the edge, it used to happen at the edge **and** with mouse inside the node. In both cases (snapped or not), `connectionStatus` is valid.

The problem is that it messes our render logic: We render the connector with a different length based on `connectionStatus` (to have the arrow fall on the edge of the node when snapped and not the center) because we assumed it was correlated to the snapping effect. 

### Platform

- OS: Manjaro KDE
- Browser: Chrome
- Version: 116.0.5845.110

**Comment by moklick** [comment]
I will check it!

**Comment by moklick** [comment]
The reason for this behaviour is that we always use the handle below the mouse when we find one. If there is none, we check if we are inside the connection radius and only if that's the case you get the snapping. This means that you should be able to display a 1x1px handle and just increase the connection radius based on the size of your nodes. Does this solve your issue?

**Comment by strblr** [comment]
Not really, given your description it should work. Each node has two invisible handles under the visible dot, just a big wider. With a connection radius that goes even further. If I understand your description correctly, it should snap to the node no matter where the mouse is on the node. Which was the case before.

**Comment by moklick** [comment]
The handle should be smaller than the node, so that you can use the connectionRadius snapping which will only be used when the mouse is not above the handle. This example shows a possible implementation: https://codesandbox.io/s/morning-sunset-xn57kz?file=/App.js:1690-1718

**Comment by strblr** [comment]
I need the source handle to be larger than the node because when starting a connection, I want to be able to start from the edge (the node itself is for dragging the node).

![CPT2309191744-829x506](https://github.com/wbkd/react-flow/assets/30187797/bdb47bf8-e9fa-4dc0-a9d1-3128d478983c)

**Comment by moklick** [comment]
This should work when only the source handle has width and height 100%, example: https://codesandbox.io/s/morning-sunset-xn57kz?file=/CustomNode.js

**Comment by moklick** [comment]
Is this fixed or still an issue?

**Comment by strblr** [comment]
I didn't manage to do it, so I simply didn't upgrade. It works well in all versions prior to 11.7.3, not sure why it broke.

**Comment by moklick** [comment]
ok, thanks for the info!

**Comment by kishan-getstarted** [comment]
Hi came across this thread while looking for the same ... is there any way I can make entire node as handler so I can drop edge anywhere ?

@moklick @strblr

**Comment by emdrayallday** [comment]
bump, same issue. I made the handle w-full h-full of the node. I can drop the line and it then connects..but the snapping effect does not happen.

It's a rectangular node, so the connection radius doesn't make sense to use.

**Comment by moklick** [comment]
Please re-open if the issue still occurs in the latest v12 release.
